### DISCUSSION TITLE
I cannot access response data if i send any statusCode other than 200

### DISCUSSION THREAD

**Discussion by 1finedev** [vercel/next.js#36930] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.15.0
      npm: 8.8.0
      Yarn: 1.22.10
      pnpm: N/A
    Relevant packages:
      next: 12.1.6
      react: 18.1.0
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

I try to send the appropriate response status code when sending response data from the api routes  but it fails to display response data unless its a 200 status code

### Expected Behavior

i expect i do not have to send a 200 status code for every response

### To Reproduce

Whenever i send a response like below i cant seem to access the response data to display the custom error message i specified, i hope i am not doing something wrong 

```js
 return res.status(400).json({
    status: "success",
    msg: "error",
  });
};
````
i only get a request failed log in the console

but with 
```js
return res.status(200).json({
    status: "success",
    msg: "error",
  });
};
```

i can easily access the response data and access the custom error message 
i need guidance 
thanks

**Comment by balazsorban44** [comment]
It's hard to tell without a reproduction, but it's likely that in your client-code, you don't handle the 400 code properly and it throws an error. Check out the MDN docs:

https://developer.mozilla.org/en-US/docs/Web/API/Response/ok
https://developer.mozilla.org/en-US/docs/Web/API/Response/status

**Comment by icyJoseph** [comment, marked as answer]
Hi,

In one of our applications, except for 3xx, family of codes, 2xx, 4xx and 5xx, use json and we can certainly decode it. Are you perhaps using `axios` client side, or some other kind of fetch client?

**Reply by 1finedev** [reply]
yes i am using axios

**Reply by icyJoseph** [reply]
Here's your answer then. Taken from [here](https://github.com/axios/axios#handling-errors):

> The request was made and the server responded with a status code
> that falls out of the range of 2xx

```js
axios.get('/user/12345')
  .catch(function (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else if (error.request) {
      // The request was made but no response was received
      // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
      // http.ClientRequest in node.js
      console.log(error.request);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log('Error', error.message);
    }
    console.log(error.config);
  });
```

Axios has special handling for non `2xx` responses. In short, `fetch` and `axios` don't have the same behavior.

**Reply by 1finedev** [reply]
thanks ill check this out

**Reply by icyJoseph** [reply]
Please do mark as answered once resolved. This was labelled as bug, but it wasn't. It's just the fetching method on the client that was not used correctly. 

I imagine quite a bunch of people use `axios` and this might be helpful along the way.

**Comment by TBytes404** [comment]
react and react-do versions should be same!

`console.log` the whole response object and see if `msg` key exists in anywhere, eg. in `error.data.message` object.

**Reply by icyJoseph** [reply]
Hi, 

That was a nice catch! I had missed that as well. Sharp eyes.
